### ISSUE TITLE
`!!MISSING: command!!` above enums/structs after opening VS Code

### ISSUE THREAD

**Issue by Aloso** [rust-analyzer/rust-analyzer#3538]
Every time I open VS Code, where a Rust project is opened, the text `!!MISSING: command!!` appears above every struct and enum, where the number of implementations should be:

![Bug screenshot](https://user-images.githubusercontent.com/15658558/76258727-6df88580-6254-11ea-91a2-69b15c50b800.png)

The bug doesn't disappear until I close the file and re-open it.

EDIT: This text only appears in one project, which uses LALRPOP. In my other projects, no text is displayed. The number of implementations only appears after closing and re-opening the file.

**Comment by Veetaha** [comment]
This is very strange. Might I ask you to share the project and `rust-analyzer` version you use to reproduce this?

**Comment by Aloso** [comment]
I'm no longer having this issue in the latest release. I think I can close this.

**Comment by Veetaha** [comment]
Feel free to reopen one you see it again. To be prepared, I'd recommend enabling
```json
{
    "rust-analyzer.trace.extension": true,
    "rust-analyzer.trace.server": "verbose",
}
```
So that you can provide more info from `Help > Toggle Developer Tools > Console` and from `Output >  Rust Analyzer Language Server Trace`

**Comment by damienstanton** [comment]
It may be worth reopening this issue: I am seeing this on the current release `0.2.489` and toolchain `stable-x86_64-apple-darwin`

<img width="144" alt="Screen Shot 2021-02-21 at 9 38 20 AM" src="https://user-images.githubusercontent.com/4822375/108628379-91e2b600-7428-11eb-9329-697eb2d479a8.png">

**Comment by bjorn3** [comment]
I also saw it a few days ago. Can't remember what I was doing though.

**Comment by damienstanton** [comment]
In my case this is occurring with a simple project with no dependencies and not in a workspace.

I tried uninstalling/reinstalling the rust-analyzer plug-in and reloading VSCode, to no avail.

I'll see if I can get trace output to get more details.

**Comment by Rrogntudju** [comment]
Same issue here but not going away. Happens only in one file in a workspace of 3 members. Maybe not related, there is another issue in the same file: the hover definition doesn't work. Here is an example from the trace:

[Trace - 17 h 49 min 20 s] Sending request 'textDocument/hover - (239)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/--snip---/Documents/VSCode/test-oidc-rs/Druid/src/main.rs"
    },
    "position": {
        "line": 15,
        "character": 87
    }
}


[Trace - 17 h 49 min 20 s] Received response 'textDocument/hover - (239)' in 1ms.
No result returned.

**Comment by Purpzie** [comment]
I'm getting this issue as well. I'll add another comment later if I can manage to catch it in the trace... it seems so random.

**Comment by damienstanton** [comment]
I can confirm that it's intermittent. In a different project, I get the error in one file but not in another in the same module.

**Comment by Veykril** [comment]
Yep looks like this appears in rust files that arent in the module tree? Every item in https://github.com/rust-analyzer/rust-analyzer/blob/0537510aef4059d5f79146a8dc2734ffdb27dc74/crates/ide_db/src/helpers/famous_defs_fixture.rs has this message as well, the file itself isnt a module of RA but is being included as a str instead.

**Comment by ppershing** [comment]
I can chime in on the "missing in the module tree" as well. For me this happens on all examples from https://github.com/rust-lang/rustlings but main code works flawlessly.

**Comment by Veykril** [comment]
Yep just checked a bit more and it is indeed the case for modules not in the tree. We create code lenses for all rust files as this creation is solely AST-based but things go wrong when they are resolved, here https://github.com/rust-analyzer/rust-analyzer/blob/f57e2f55984758a83644b852a4cc47e0b27945df/crates/rust-analyzer/src/handlers.rs#L1131-L1138
This calls `resolve_annotation` which in turn `None`s the data here https://github.com/rust-analyzer/rust-analyzer/blob/f57e2f55984758a83644b852a4cc47e0b27945df/crates/ide/src/annotations.rs#L127-L129 due to `goto_implementation` returning `None` since we have no HIR for this file.
This then causes us to return a CodeLens with no command set on resolve here which causes VSCode to show a `!!MISSING: command!!` as its missing a command https://github.com/rust-analyzer/rust-analyzer/blob/f57e2f55984758a83644b852a4cc47e0b27945df/crates/rust-analyzer/src/to_proto.rs#L906-L930

**Comment by lnicola** [comment]
@Veykril can we close this?

**Comment by Veykril** [comment]
Yup, this should be fixed in #8015.

**Comment by boehs** [comment]
Reappearing on nightly

![](https://e.reeee.ee/xjQTZS.png)

Using `"rust-analyzer.updates.channel": "nightly"` in my vscode config
